### PR TITLE
Persistence in optimizer

### LINKED ISSUE #329: Optimizer Persistence
--------
Issue Description
--------
##### What did you expect to see happen?
It would be nice if the optimizer could double-check its solution a number of times before accepting convergence.

##### What did you see instead?
Any convergence is accepted as job completed.

##### Do you have a suggested fix for the development team?
Implement persistence.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? __not a defect__
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) __not a defect__
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)? __not a defect__
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? __not a defect__
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? __N/A__

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #329.

##### What are the significant changes in functionality due to this change request?
Implements rudimentary persistence in the optimizer.  Any consecutive convergence will quality as an acceptable contribution to the persistence.

### CODE REVIEW

File: tests/framework/Optimizers/Persistence/parabola.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 np.random.seed(42)
 
 def base(values):
-  return -sum(l*l for l in values)+1.
+  return -sum(x**2 for x in values)+1.
```
Comment (PaulTalbot-INL) [inline tests/framework/Optimizers/Persistence/parabola.py]: Approved.

File: tests/framework/Optimizers/persistence.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>framework/Optimizers.Persistance</name>
+    <author>talbpaul</author>
+    <created>2017-08-06</created>
```
Comment (alfoa) [inline tests/framework/Optimizers/persistence.xml]: 09 :D

File: tests/framework/Optimizers/persistence.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    <description>
+      This test runs the optimization on a inverse parabola with a persistance greater than 1 to
+      demonstrate mechanics.
+    </description>
+    <analytic>
+      This tests uses the inverse parabola for optimization. The inverse parabola is documented
+      in the analytic tests, and has an optimal max point at $x=0$.
+    </analytic>
+    <revisions>
+      <revision author="talbpaul" date="2017-05-14">creation</revision>
```
Comment (alfoa) [inline tests/framework/Optimizers/persistence.xml]: I do not think this is needed.

File: tests/framework/Optimizers/Persistence/parabola.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Computes an inverse parabola with maximum value at x=0 of 1.0.
+#
+import numpy as np
+
+#static seed
+np.random.seed(42)
+
+def base(values):
+  return -sum(x**2 for x in values)+1.
```
Comment (maljovec) [inline tests/framework/Optimizers/Persistence/parabola.py]: I made a minor change to this line as the online editor led me to believe the variable `l` was actually a `1`.

**Pull request merged**